src/redux/reducers/userReducer.ts

```diff
@@ -1,20 +1,23 @@
 import {createReducer} from '@reduxjs/toolkit';
+import {User} from "./User";
 
 const initialState = {
     isAuthenticated: false,
     loading: false,
     isLoading: false,
-    user: {},
+    user: {} as User,
```

Comment:
It's not safe. We're trying to fool the TS compiler, which makes our code less reliable. We're saying `{}` is a user, but is we do `state.user.name.toUpperCase()` we're getting a runtime error. It could have been caught early in development if we didn't abuse TS here